aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group/Program.cs

```diff
@@ -0,0 +1,201 @@
+using Data;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+var connection = new SqliteConnection("DataSource=:memory:");
+connection.Open();
+// Add InMemoryDatabase to the container.
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+{
+    options.UseSqlite(connection);
+});
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    // localhost:{port}/swagger
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.MapGet("/", () => "Hello World!");
+
+// todo endpoints
+var todos = app.MapGroup("/todos").WithTags("Todo Endpoints");
+todos.MapGet("/", GetAllTodos);
+todos.MapGet("/{id}", GetTodo);
+todos.MapPost("/", CreateTodo).AddRouteHandlerFilter((context, next) =>
+{
+    if (context.HttpContext.Request.ContentLength == 0)
+    {
+        context.HttpContext.Response.StatusCode = 400;
+        return new ValueTask<object?>(Results.BadRequest(new { Message = "Request body is too empty" }));
```

Comment:
Nit: Making this async can get rid of the `new ValueTask<object?>(` and [ValidationProblem](https://github.com/dotnet/aspnetcore/blob/896ba363baad187982047ce80d6901d352f75b19/src/Http/samples/MinimalSample/Program.cs#L84) might be a better choice than a plain `BadRequest`. Also, we should try using `TypedResults` over `Results` anywhere we can.

```suggestion
        return TypedResults.ValidationProblem(errors);
```